website/scripts/ros/ros_traffic_signal.js

```diff
@@ -1,67 +1,75 @@
+                            let signal_state = event_ele.event_state.movement_phase_state;
+                            let remaining_time = event_ele.timing.min_end_time;
+                            //Prevent repeating the same state
+                            if(signalState !=signal_state)
+                            {
+                                signalState = signal_state;
+                                    switch(signalState)
+                                    {
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.protected_movement_allowed:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('green',remaining_time));
+                                            break;
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.stop_and_remain:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('red',remaining_time));
+                                            break;
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.protected_clearance:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('yellow',remaining_time));
+                                            break;
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.permissive_movement_allowed:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('flash_green',remaining_time));
+                                            break;
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.permissive_clearance:
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.caution_conflicting_traffic:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('flash_yellow',remaining_time));
+                                            break;
+                                        case TRAFFIC_SIGNAL_PHASE_STATE.stop_then_proceed:
+                                            $('.traffic-signal-col').append(updateTrafficSignal('flash_red',remaining_time));
+                                            break;
+                                        default:
+                                            $('.traffic-signal-col').html('');
+                                            $('.traffic-signal-col').append(updateTrafficSignal('',''));
+                                            console.error("Traffic signal state is invalid");
+                                            break;
+                                    } 
+                                    //set back to black after 5 seconds.
+                                    if(!isTimeerSet)
+                                    {
+                                        isTimeerSet=true;
+                                        setTimeout(function(){
```

Comment:
This timer clears everything every 5 seconds. Why is that desirable? Wouldn't you only want the timer to clear if no data was received for 5 seconds?

Reply:
What is idea sec?

Reply:
Is the topic being listened only publishes once or multiple times? If multiple times, this 1 second is ok. Also have we implemented the count down per phase?

Reply:
pub at 5HZ

Reply:
yes, the count down is the min_end_time